cpp/src/arrow/array/array_base.h

```diff
@@ -176,7 +176,7 @@ class ARROW_EXPORT Array {
   Status ValidateFull() const;
 
  protected:
-  Array() : null_bitmap_data_(NULLPTR) {}
+  Array() : data_(NULLPTR), null_bitmap_data_(NULLPTR) {}
```

Comment:
There are multiple `Conditional jump or move depends on uninitialised value(s)` from `Array.IsNull` and `Array.IsValid` but it doesn't seem to solve it.

Reply:
Yes, this change is not useful as `shared_ptr` is always initialized.

Reply:
I initialized as ArrayData because IsNull references `data_->offset`

Reply:
I don't think that would solve anything. As I said, `shared_ptr` is always initialized (to nullptr).